### DISCUSSION TITLE
Error in Plotting in R

### DISCUSSION THREAD

**Discussion by eeinnew** [teamalgoritma/community#87] [Q&A] [answered]
Hello, I would like to ask about my error in plotting area chart with ggplot2
<img width="960" alt="image" src="https://user-images.githubusercontent.com/81936292/167318970-40ddf827-c2e3-4b83-aa33-9f6e81bad1e0.png">
<img width="960" alt="image" src="https://user-images.githubusercontent.com/81936292/167319076-40814817-4f53-4a00-bb00-ac58685d5bc4.png">
Firstly, as you can see, the geom_point and the geom_area is not the same(while it supposed to be the same) for ex = they both should be on 10.06 (Geom_point) is the correct one. (as the highest values is  10.06)
<img width="960" alt="image" src="https://user-images.githubusercontent.com/81936292/167319022-0aa5fe97-01bb-49ed-9994-ab7275e2a33e.png">

Lastly, I have stated the legend_position at the bottom of the graph but it keeps appearing on the right side

**Comment by VictorNugraha** [comment, marked as answer]
Hello, @eeinnew !

Actually, there are no errors occur in the plot that you has been made. 

The reason why _geom_area_ position is not the same with actual value represented by _geom_point_ is because the _geom_area_ default position is _**stacked**_. In order to make the _geom_area_ the same with _geom_point_ , you can change the default position by adding parameter _**position = "identify"**_ into _geom_area_ function. The caveat here, you might need to add parameter **_alhpa = .3_** at _geom_area_ in order to prevent some of the values ​​may be hidden behind a bigger value.

And for the legend position, when you want to change the legend position for interactive plotting, you have to adjust it using function called _**layout**_. Here are the example of the code: 
```
ggplotly(plot) %>% 
  layout(legend = list(orientation = "h", # this parameter used to control legend orientation, it can filled with "h" (horizontal) or "v" (vertical)
                                   x = 0.4,  # this parameter used to move the legend position horizontally
                                   y = -0.2)) # this parameter used to move the legend position vertically
```

That's the information I can convey, I hope it answers the questions given and good luck! :D

**Reply by eeinnew** [reply]
oh yes, it works. anyway is there any ways to change the legend sub-name > for ex : female un employment, male unemployment because the name is too long (as it follows what is stated in the csv data)

**Reply by VictorNugraha** [reply]
Yes, you can! You can add a parameter called _**labels**_ at  _geom_area_ function. Parameter _**labels**_ can help us to change the actual name from our data frame into specific name to be shown at our legend plot. Here are the example of the code:

`ggplot(...) +
     geom_area(.... , labels = c("Desired Name 1", "Desired Name 2")
`

**Comment by eeinnew** [comment]
sorry but it does not work on me
![image](https://user-images.githubusercontent.com/81936292/167386430-473aff10-a597-40f0-9cba-e16ad891d96c.png)
![image](https://user-images.githubusercontent.com/81936292/167386454-c1d86049-f732-4775-bd92-ac82838b1992.png)
I tried it too on goem_bar ( but it ignore this label) and nothing happen

**Reply by VictorNugraha** [reply]
If you don't mind, you can change the actual data into your desired name. You can use function _**fct_recode()**_ from **_library(forecast)_** and function _**mutate()**_ from _**library(dplyr)**_.

Let's say, I have a data frame like down below

![image](https://user-images.githubusercontent.com/28379954/167392945-80fa1c09-6b2a-40ba-a7b0-2600eaad45e6.png)

From the data frame, I want to change gender value from F -> Female and M -> Male.

Here are the example of the code:

```
df <- df %>%  
  mutate(gender = fct_recode(.f = gender, # Column name that the value we would like to change
                             "Female" = "F", # "Desired Value" = "Initial Value"
                             "Male" = "M" # "Desired Value 2" = "Initial Value 2")) 
```

And the new data frame will be look like this,

![image](https://user-images.githubusercontent.com/28379954/167395066-16705741-64bf-4f0d-9d58-9f0a6ccb4e3c.png)

Hopefully this method will solved your problem! :D

**Reply by eeinnew** [reply]
sorry, but my data type is double, not factor so it is not working :(

**Reply by VictorNugraha** [reply]
Is it possible to change your data type into factor data type? In my humble opinion that column (_variable column_) is more suitable to have factor data type, since from my observation based on your result, that column consist only two repetitive value (_unemployment_female.._ & _unemployment_male.._).

Please correct me if I did misinterpretation your question :)

**Comment by eeinnew** [comment]
sorry my mistake. 
However, when i try to knit in flexdashboard, i dont know why it is error (I have called the library)
![image](https://user-images.githubusercontent.com/81936292/167536953-c8a00d87-8481-4759-b700-6f1d38c75249.png)

When I run the chunks in r mark down, it works, but having trouble when being rendered

**Comment by eeinnew** [comment]
seems the library should be in tidyverse, and it works!
anyway thankyou for your assistance pak Victor!
